### PR TITLE
First take on a dev setup

### PR DESCRIPTION
**Pull request by ronilan**
This pull request creates a Mac dev environment.

1. Modifies `.gitignore` (a6f3fe2).
2. Adds Mocha as local dev dependency (af24dfd).
3. Everything related to dev is in dev directory (ef8d259)
  - Dev container has build tools, docker, docker compose and multiple versions of node installed as well as specific Oracle and PostgreSQL related installs.
 - The dev container is setup with multiple connected services using docker compose.
 - There is simple way to get local "one off" containers.
 - There is an easy way to work with "formalized" dev reop.

4. Four npm commands launch four shell scripts (e53e65d):
a. `npm run dev` - starts and ubuntu based dev environment defined in a docker compose. Code is mounted. This is where work should be done.
b. `npm run dev:oneoff` - starts a "one off" dev environment. code is copied. This is for testing specific node/os things.
c. `npm run dev:repo:reset` - runs a set of git operations that "resets" the dev repo to match the main repo.
d. `npm run dev:reinstall` - removed node_modules and runs npm install. To be used after switching node versions.

The setup is similar to the one used in upstream repo https://github.com/appoptics/appoptics-bindings-node/pull/63

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,32 @@
...
-node_modules
+# anything, anywhere starting with __ or ~~
+__*
+~~*
+
+# Local development
+*.env
+*.dev
+
+# Apple
```
Comment (prettycoder) [inline .gitignore:9]: I recently discovered global gitignore https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files#configuring-ignored-files-for-all-repositories-on-your-computer
imho it is a cleaner way to exclude machine specific files

File: dev/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,60 @@
+FROM ubuntu:20.04
+SHELL ["/bin/bash", "-c"]
+
+# tools
+RUN su - \
```
Comment (prettycoder) [inline dev/Dockerfile:5]: I never need su or sudo on an ubuntu image, why do you need it?

File: dev/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+                g++ \
+                python \
+                make \
+                postgresql-server-dev-12 \
+                zip \
+                unzip
+
+# install docker (inception)
+RUN curl -fsSL https://get.docker.com -o get-docker.sh && \
+    sudo sh get-docker.sh
```
Comment (prettycoder) [inline dev/Dockerfile:21]: sudo needed?

File: dev/repo/reset.sh [hunk 4]
```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
```
Comment (prettycoder) [inline dev/repo/reset.sh:1]: interesting ... why is this needed?

File: package.json [hunk 5]
```diff
@@ -14,7 +14,11 @@
...
     "gen-docs": "cat lib/index.js lib/api.js lib/span.js lib/event.js > api-docs-temp.js && jsdoc2md -f api-docs-temp.js > guides/api.md && rm api-docs-temp.js",
     "install-deps": "./deps.sh",
-    "lint": "eslint --quiet lib/*.js lib/**/*.js test/*.js test/**/*.js prepack.js"
+    "lint": "eslint --quiet lib/*.js lib/**/*.js test/*.js test/**/*.js prepack.js",
+    "dev": "dev/start.sh",
+    "dev:oneoff": "dev/oneoff.sh",
+    "dev:repo:reset": "dev/repo/reset.sh",
+    "dev:reinstall": "rm -rf node_modules && npm install --unsafe-perm"
   },
   "author": "Bruce A. MacNaughton <bruce@solarwinds.cloud>",
```
Comment (prettycoder) [inline package.json:23]: do you wanna add your name?

**Review by prettycoder — APPROVED** [review]
lgtm

just one question about sudo/su in the Dockerfile

File: package.json (hunk 5 above)
Comment (ronilan) [reply package.json:23]: A lot of config mods incoming... will add it sometime...

File: dev/repo/reset.sh (hunk 4 above)
Comment (ronilan) [reply dev/repo/reset.sh:1]: See other repo.

File: dev/Dockerfile (hunk 3 above)
Comment (ronilan) [reply dev/Dockerfile]: Naively taken from [docs](https://docs.docker.com/engine/install/ubuntu/#install-using-the-convenience-script). Most likely not needed.

Thinking things over, since currently, neither Docker nor Docker Compose are needed, and since there are other problems with such inception, the whole thing was removed. 

Re-review will be needed (repo setting nullify reviews on code change...).

File: .gitignore (hunk 1 above)
Comment (ronilan) [reply .gitignore:9]: I'm actually pulling to the other direction (for now), avoiding anything global (e.g. packages). The goal is to get to a point where node development can be done on a factory reset Mac with minimal configuration (git + docker +  ??? ). There is some more WIP here as there are more repos involved (testing).

File: dev/Dockerfile (hunk 2 above)
Comment (ronilan) [reply dev/Dockerfile]: removed.

**Review by prettycoder — APPROVED** [review]
looks good
